node/src/components/consensus/protocols/highway.rs

```diff
@@ -356,6 +359,40 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
     pub(crate) fn median_round_exp(&self) -> Option<u8> {
         self.highway.state().median_round_exp()
     }
+
+    fn drop_dependent_vertices<It>(&mut self, vertices: It)
+    where
+        It: IntoIterator<Item = Dependency<C>>,
+    {
+        // collect the vertices that depend on the ones we got in the argument and their senders...
+        let (senders, dependent_vertices): (HashSet<I>, Vec<Dependency<C>>) = vertices
+            .into_iter()
+            .flat_map(|vertex| self.vertex_deps.remove(&vertex))
+            .flatten()
+            .map(|(sender, pvv)| (sender, pvv.inner().id()))
+            .unzip();
+        // ...and drop them, too
+        self.drop_dependent_vertices(dependent_vertices);
+        // also, drop all vertices from the senders - if they sent us something dependent on an
+        // invalid vertex, they can't be trusted
+        self.drop_by_senders(senders);
```

Comment:
Actually, maybe this is dangerous:
The invalid vertex could be invalid "evidence against Eric", even though Eric actually _is_ evil. Then we'd drop all the other vertices by correct validators here, just because they also depend on evidence against Eric.

Reply:
As discussed on Zoom, we will not drop vertices that depend on invalid evidence or endorsements - as the dependency can still be satisfied by correct evidence or endorsements. We can still drop vertices dependent on invalid units (formerly votes), though.

Reply:
Maybe add a comment so the reader knows why we're filtering with `is_unit` here.